### DISCUSSION TITLE
Different context for each resolver

### DISCUSSION THREAD

**Discussion by johannbuscail** [MichalLytek/type-graphql#1137] [Q&A]
I use Neo4j Driver to connect my app to my database. 
Before each resolver, I would like to create a session and close it once the resolver ends. 
I tried creating a middleware to put the session in the context:
```ts
export const GraphQLCreateSession: MiddlewareFn<RequestContext> = async ({ context }, next) => {
	const session = context.driver.session(databaseOpts);
	context.session = session;
        console.log("creating session")
	try {
		const result = await next();
		return result;
	} finally {
                console.log("closing session")
		await session.close();
	}
};
```
I was thinking the context would be different for each resolver, but it's not.
For each resolver, it overrides the session that was created before.
At then, it only closes the last session that was created in the context.

**Comment by MichalLytek** [comment]
Context is execution context for the whole GraphQL document you send by client and execute on the server.
If you need something for each resolver, just use a local variable.
